src/otx/cli/utils/help_formatter.py

```diff
@@ -0,0 +1,220 @@
+"""Custom Help Formatters for OTX CLI."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import re
+import sys
+from typing import TYPE_CHECKING, Iterable
+
+from jsonargparse import DefaultHelpFormatter
+from rich.markdown import Markdown
+from rich.panel import Panel
+from rich_argparse import RichHelpFormatter
+
+if TYPE_CHECKING:
+    import argparse
+
+    from rich.console import RenderableType
+
+
+BASE_ARGUMENTS = {"config", "print_config", "help", "engine", "model", "model.help", "task"}
+ENGINE_ARGUMENTS = {"data_root", "engine.help", "engine.device", "engine.checkpoint", "engine.work_dir"}
+REQUIRED_ARGUMENTS = {
+    "train": {
+        "data",
+        "optimizer",
+        "scheduler",
+    }.union(BASE_ARGUMENTS).union(ENGINE_ARGUMENTS),
```

Comment:
Not sure what's the convention here but I find the following easier to read
```python
{
...,
*BASE_ARGUMENTS,
*ENGINE_ARGUMENTS
}
```
But feel free to ignore this. It is a personal preference.

Reply:
Thanks! Done - https://github.com/openvinotoolkit/training_extensions/commit/ebc0361b443568eca751738fd744bbb8b140c59a